tests/aiohttp/test_graphql_transport_ws.py

```diff
@@ -955,3 +955,38 @@ async def test_single_result_duplicate_ids_query(aiohttp_client):
 
         await ws.close()
         assert ws.closed
+
+
+async def test_injects_connection_params(aiohttp_client):
+    app = create_app()
+    aiohttp_app_client = await aiohttp_client(app)
+
+    async with aiohttp_app_client.ws_connect(
+        "/graphql", protocols=[GRAPHQL_TRANSPORT_WS_PROTOCOL]
+    ) as ws:
+        await ws.send_json(ConnectionInitMessage(payload="echo").as_dict())
```

Comment:
I assume this test is passing just fine, but technically the payload is required (by both ws protocols) to be a JS object / Python dict or null/None. We should adjust the test accordingly even tho just having a string works :) I'm confused why mypy is not complaining since `ConnectionInitMessage` is typed accordingly.

Reply:
I agree that tests should assume that the provided `connection_params` is passed as a dict. As a result of this assumption, should we enforce that `connection_params` be a dict, _i.e._ during the `handle_connection_init` test the typing of the object and reject it if it isn't dict-like?

Reply:
I have implemented a check on the payload variable to enforce it as a dict or None, let me know if you think it's welcome :grin:

Reply:
> should we enforce that connection_params be a dict

I think we should, but for all `payload` objects. The [graphql-ws-transport spec](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#invalid-message) specifies what should happen when a message is invalid. The [graphql-ws protocol](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) doesn't appear to be to explicit about it. (note that the links are correct, the protocol repository names do not match their protocols name).